forge/routes/api/project.js

```diff
@@ -361,7 +361,55 @@ module.exports = async function (app) {
         }
     }, async (request, reply) => {
         let changed = false
-        if (request.body.stack) {
+        if (request.body.changeProjectType === true) {
```

Comment:
This end point is somewhat overloaded, but it does already let you set the ProjectType of the project **only if** the project doesn't already have a ProjectType set. This was added when we introduced ProjectTypes as a one-off task for users to set the type of their project. This is why the handling for that will bail out early if the Project already has a ProjectType set.

I'm wondering if, rather than introduce a `changeProjectType` flag when we don't have equivalents for the other things you can change, this logic should be moved down to the existing handling when `body.projectType` is set (line 434 of the proposed changes).

That block then has two branches - one for setting ProjectType initially, and one for changing ProjectType.

We should look at the data on production, but handling projects without a ProjectType is very much a legacy edge case that would only apply to FF instances that have been running since before ProjectTypes were introduced.

Reply:
> I'm wondering if, rather than introduce a changeProjectType flag when we don't have equivalents for the other things you can change, this logic should be moved down to the existing handling when body.projectType is set (line 434 of the proposed changes).

Essentially, there was the original "no project type" code that was rather implicit in the fact that if `body` supplied a `projectType`. 

With the introduction of Change Project Type, the `body` needs to contain both `projectType` AND `stack` in the `body` - so the explicit `changeProjectType` was introduced to make this 100% explicit as to intention rather than inferring intention from values in body. It is much more deliberate than being present or not.   

Also, since the first part of the logic would capture execution (because `body.stack` is needed in a project type change) we would have to have some `if (stack && !projectType)` logic in the first part of the handler (i.e. less explicit intention)